tests/Integration/Adapter/ConfigurationTest.php

```diff
@@ -0,0 +1,230 @@
+                        'value' => 'parent_group_value',
+                        'shopConstraint' => ShopConstraint::shopGroup(1),
+                    ],
+                ],
+                [
+                    'key' => 'parent_group_key_2',
+                    'default' => false,
+                    'shopConstraint' => ShopConstraint::shop(2),
+                ],
+                'parent_group_value',
+                false,
+            ],
+            // try getting a non existing value for shop 2, with is strict = true => should not inherit from parent group
+            [
+                [
+                    [
+                        'key' => 'parent_group_key_3',
+                        'value' => 'parent_group_value',
+                        'shopConstraint' => ShopConstraint::shopGroup(1),
+                    ],
+                ],
+                [
+                    'key' => 'parent_group_key_3',
+                    'default' => false,
+                    'shopConstraint' => ShopConstraint::shop(2, true),
+                ],
+                null,
+                false,
+            ],
+            // try getting a non existing value for all shop, with is strict = true => should return null
+            [
+                [],
+                [
+                    'key' => 'does_not_exist',
+                    'default' => false,
+                    'shopConstraint' => ShopConstraint::allShops(true),
+                ],
+                null,
+                true,
+            ],
+        ];
+    }
+
+    private function initMultistore(): void
+    {
+        // we want to execute this only once for the whole class
+        $flag = $this->configuration->get('CONFIGURATION_INTEGRATION_TEST_FLAG');
+
+        if ($flag === null) {
+            // activate multistore
+            $this->configuration->set('PS_MULTISHOP_FEATURE_ACTIVE', 1);
+
+            // add a shop in existing group
+            $shopGroup = $this->entityManager->find(ShopGroup::class, 1);
+            $shop = new Shop();
+            $shop->setActive(true);
+            $shop->setIdCategory(2);
+            $shop->setName('test_shop_2');
+            $shop->setShopGroup($shopGroup);
+            $shop->setColor('red');
+            $shop->setThemeName('classic');
+            $shop->setDeleted(false);
+
+            $this->entityManager->persist($shop);
+            $this->entityManager->flush();
+
+            // activate flag
+            $this->configuration->set('CONFIGURATION_INTEGRATION_TEST_FLAG', 1);
+        }
+    }
+
+    public function cleanDb(): void
+    {
+        //remove newly created shop
+        $shop = $this->entityManager->getRepository(Shop::class)->findOneBy(['name' => 'test_shop_2']);
+        $this->entityManager->remove($shop);
+        $this->entityManager->flush();
+        // remove multistore feature enabling configuration
+        $this->configuration->remove('PS_MULTISHOP_FEATURE_ACTIVE');
```

Comment:
```suggestion
        $this->configuration->set('PS_MULTISHOP_FEATURE_ACTIVE', 0);
```

The configuration must not be removed, just reset to 0